### ISSUE TITLE
Moving items in Zones & Devices lets Living room disappear

### ISSUE THREAD

**Issue by Luukvis** [athombv/homey#174]
I moved al zones (living room, bedroom etc) that where under "Ground Floor" to "Home" to make the view easier. Then I removed "Ground Floor". What happened is that Living room disappeared in the list (after going back to settings e.g.). 
Al devices that were created under Living Room are still visible in the Flow items, but can't be used anymore. Recreating Living room does not help. All the created devices are gone in the Zones & Devices view, but visible in the Flow items. Somehow there is a mismatch in the database.

**Comment by WeeJeWel** [comment]
That is really odd. Could you execute this in Google Chrome's console (Right click somewhere -> Inspect element -> Tab 'Console'):

```
api('GET', '/manager/zones/', function( err, zones ){ console.log(JSON.stringify(zones, false, 4))})
```

And expand all output, and paste it here.

Example output:

```
[
    {
        "id": "9919ee1e-ffbc-480b-bc4b-77fb047e9e68",
        "parent": false,
        "name": "Home",
        "icon": "home",
        "index": 1
    },
    {
        "id": "37782dac-6d40-40a2-a0a4-17a0dd05003a",
        "name": "Basement",
        "parent": "9919ee1e-ffbc-480b-bc4b-77fb047e9e68",
        "index": 6,
        "icon": "stairs-down"
    },
    {
        "id": "b840a982-a3a7-4513-9d96-902d81128f5c",
        "name": "First floor",
        "parent": "9919ee1e-ffbc-480b-bc4b-77fb047e9e68",
        "index": 5,
        "icon": "stairs-up"
    },
    {
        "id": "6a48add5-7349-4ff3-84b9-812e2d706e50",
        "name": "Attic",
        "parent": "9919ee1e-ffbc-480b-bc4b-77fb047e9e68",
        "index": 7,
        "icon": "roof"
    },
    {
        "id": "9eb2975d-49ea-4033-8db0-105a3e982117",
        "name": "Living Room",
        "parent": "855113f1-f488-4223-b675-2f01270f573e",
        "index": 1,
        "icon": "living"
    },
    {
        "id": "2ab5a003-0120-4c01-80f2-a237dcf4ba14",
        "name": "Kitchen",
        "parent": "9919ee1e-ffbc-480b-bc4b-77fb047e9e68",
        "index": 2,
        "icon": "kitchen"
    },
    {
        "id": "ddc3cdaa-8172-4c4a-97c9-22d8ec462972",
        "name": "Bedroom",
        "parent": "b840a982-a3a7-4513-9d96-902d81128f5c",
        "index": 1,
        "icon": "bed"
    },
    {
        "id": "5204dc44-8e9c-49e7-aed8-9d5c93330126",
        "name": "Garden",
        "parent": "9919ee1e-ffbc-480b-bc4b-77fb047e9e68",
        "index": 8,
        "icon": "garden"
    },
    {
        "id": "855113f1-f488-4223-b675-2f01270f573e",
        "name": "Ground floor",
        "parent": "9919ee1e-ffbc-480b-bc4b-77fb047e9e68",
        "index": 3,
        "icon": "default"
    },
    {
        "id": "8a6928ea-26b2-4922-9ead-d20ee6695dd3",
        "name": "Bathroom",
        "parent": "b840a982-a3a7-4513-9d96-902d81128f5c",
        "index": 3,
        "icon": "bed"
    },
    {
        "id": "1815c884-af06-4d53-a2c1-6f4c77e9eb4e",
        "name": "Study",
        "parent": "b840a982-a3a7-4513-9d96-902d81128f5c",
        "index": 2,
        "icon": "books"
    },
    {
        "id": "478cfd92-6475-4214-80d9-27125e39863a",
        "name": "Hallway",
        "parent": "9919ee1e-ffbc-480b-bc4b-77fb047e9e68",
        "index": 4,
        "icon": "default"
    }
]
```

**Comment by Luukvis** [comment]
The results:
[
    {
        "id": "9919ee1e-ffbc-480b-bc4b-77fb047e9e68",
        "parent": false,
        "name": "Home",
        "icon": "home",
        "index": 1
    },
    {
        "id": "9eb2975d-49ea-4033-8db0-105a3e982117",
        "name": "Living Room",
        "parent": "855113f1-f488-4223-b675-2f01270f573e",
        "index": 1,
        "icon": "living"
    },
    {
        "id": "2ab5a003-0120-4c01-80f2-a237dcf4ba14",
        "name": "Kitchen",
        "parent": "9919ee1e-ffbc-480b-bc4b-77fb047e9e68",
        "index": 4,
        "icon": "kitchen"
    },
    {
        "id": "ddc3cdaa-8172-4c4a-97c9-22d8ec462972",
        "name": "Bedroom",
        "parent": "9919ee1e-ffbc-480b-bc4b-77fb047e9e68",
        "index": 5,
        "icon": "bed"
    },
    {
        "id": "8a6928ea-26b2-4922-9ead-d20ee6695dd3",
        "name": "Bathroom",
        "parent": "9919ee1e-ffbc-480b-bc4b-77fb047e9e68",
        "index": 7,
        "icon": "shower"
    },
    {
        "id": "1815c884-af06-4d53-a2c1-6f4c77e9eb4e",
        "name": "Study",
        "parent": "9919ee1e-ffbc-480b-bc4b-77fb047e9e68",
        "index": 3,
        "icon": "books"
    },
    {
        "id": "478cfd92-6475-4214-80d9-27125e39863a",
        "name": "Hallway",
        "parent": "9919ee1e-ffbc-480b-bc4b-77fb047e9e68",
        "index": 6,
        "icon": "default"
    },
    {
        "id": "1c8d6762-2dfe-4e0c-b5d9-7237db5c1f55",
        "name": "Dining",
        "parent": "9919ee1e-ffbc-480b-bc4b-77fb047e9e68",
        "index": 2,
        "icon": "default"
    }
]

And this is what I see on my screen: 
![missinglivingroom](https://cloud.githubusercontent.com/assets/16554361/13053851/f1ed7db8-d406-11e5-8beb-f5b2011a8354.JPG)

**Comment by WeeJeWel** [comment]
I see! Fixed it in 0.8.18.

To fix your zones, execute in the Chrome console:

```
api('PUT', '/manager/zones/9eb2975d-49ea-4033-8db0-105a3e982117', { parent: '9919ee1e-ffbc-480b-bc4b-77fb047e9e68' })
```

My apologies for the inconvenience :D
